### PR TITLE
hack: script for extracting the kubeconfig for a clusterdeployment

### PR DESCRIPTION
**Pull request by staebler**
This is something that I do often enough and that is a long enough command that I thought it would be beneficial to create a script for.

### CODE REVIEW

File: hack/get-kubeconfig.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+usage(){
+	echo "Usage: $0 [CLUSTERDEPLOYMENT_NAMESPACE/]CLUSTERDEPLOYMENT_NAME"
+	exit 1
+}
+
+
+# Get the name and namespace for the clusterdeployment to duplicate.
+CD_NAME_ARG=${1?missing ClusterDeployment name}
+CD_NAME_ARR=(${CD_NAME_ARG//// })
```
Comment (gregsheremeta) [inline hack/get-kubeconfig.sh]: how does this work? i'm not great at bash

**Comment by gregsheremeta** [comment]
looks good to me. Hopefully #823 beats this, though :)

**Comment by staebler** [comment]
> looks good to me. Hopefully #823 beats this, though :)

I'll hold this to wait for #823. This PR will be easy to rebase if needed.

/hold

File: hack/get-kubeconfig.sh (hunk 1 above)
Comment (staebler) [reply hack/get-kubeconfig.sh]: * `{string/substring/replacement}` replaces the first occurrence of *substring* with *replacement*
* `(string1 string2 string3)` creates an array with the elements in the list.

So, `(${CD_NAME_ARG//// })` replaces the first forward slash `/` in `CD_NAME_ARG` with a space then creates an array where the first element is everything before the new space and the second element is everything after the new space.

**Comment by dgoodwin** [comment]
LGTM and will be handy.

**Pull request merged**